### PR TITLE
Implement Resource limits and reserves Modules

### PR DESCRIPTION
**Pull request by apavanello**
PR ref #103 

Add two new lib files (`library/dokku_resource_limit.py` and `library/dokku_resource_reserve.py`) and update Readme

### CODE REVIEW

File: library/dokku_resource_reserve.py
Comment (ltalirz) [inline library/dokku_resource_reserve.py:133]: I see that it can be useful to have the output of the command available in parsed form - I'd just like to note that this list will need to be kept up to date with new dokku releases (if new resources are added).

Can the parsing be made agnostic of this list?

**Review by ltalirz — COMMENTED** [review]
thanks a lot for the contribution @apavanello, very welcome!

Please see below a few comments - I noticed too late that I made my suggestions in the README, i.e. you'll need to move them to the source file (sorry).

It would be nice to add a test of this new functionality somewhere [here](https://github.com/dokku/ansible-dokku/blob/e3e43f377292047a63ef08665840f49f1cd9484c/molecule/default/verify.yml#L31) but for this we first need an app installed during the `verify` stage. Unfortunately, my PR https://github.com/dokku/ansible-dokku/pull/98 is blocked by a longstanding molecule bug and there doesn't seem to be any action on it... until that point no need to add a test

**Comment by apavanello** [comment]
Hi,

Maked changes on source to create a correct `README` and maked a `allowed_keys` to be agnostic, in fact i don't need  check the output of `dokku --quiet resource:limit myapp` because the output will have only two returns, the full list of resources (even when no have limits/reserves) and the error (when the app doesn't exist), so i use the current output to generate the `allowed_keys` list,  when dokku add new resources no need to change the list

In test subject, i created a branch in my repo and added some tests and all have pass (you can check in my github actions https://github.com/apavanello/ansible-dokku/actions/runs/818794950), when #98 be fixed i can make another PR to implement the resources tests
 

Sorry for english mistakes,i still learning the language

File: library/dokku_resource_limit.py
Comment (ltalirz) [inline library/dokku_resource_limit.py:163]:
```suggestion
            meta["error"] = "Unknown resource {0}, choose one of: {1}".format(
```

File: library/dokku_resource_reserve.py
Comment (ltalirz) [inline library/dokku_resource_reserve.py:40]:
```suggestion
      - The state of the resource reservations
```

File: library/dokku_resource_reserve.py
Comment (ltalirz) [inline library/dokku_resource_reserve.py:163]:
```suggestion
            meta["error"] = "Unknown resource {0}, choose one of: {1}".format(
```

**Review by ltalirz — APPROVED** [review]
> in fact i don't need check the output of `dokku --quiet resource:limit myapp` because the output will have only two returns, the full list of resources (even when no have limits/reserves) and the error (when the app doesn't exist), so i use the current output to generate the `allowed_keys` list, when dokku add new resources no need to change the list

great!

> In test subject, i created a branch in my repo and added some tests and all have pass (you can check in my github actions https://github.com/apavanello/ansible-dokku/actions/runs/818794950)

Great, thanks for this!

> when #98 be fixed i can make another PR to implement the resources tests

Cheers, I think I started to get an idea of where the `molecule` issue comes from. Will ping you once this is done (but we will merge this PR nevertheless).


> Sorry for english mistakes,i still learning the language

No worries!



The changes look good, just some final requests.

**Comment by apavanello** [comment]
@ltalirz  all done,  changes sugested are maded.

File: library/dokku_resource_limit.py
Comment (ltalirz) [inline library/dokku_resource_limit.py:163]:
```suggestion
                k, list(report.keys())
```

File: library/dokku_resource_reserve.py
Comment (ltalirz) [inline library/dokku_resource_reserve.py:162]:
```suggestion
                k, list(report.keys())
```

**Review by ltalirz — APPROVED** [review]

**Pull request merged**
